### DISCUSSION TITLE
Capacitor targetSdkVersion and android X

### DISCUSSION THREAD

**Discussion by MikhailFokanov** [HaylLtd/cordova-plugin-background-geolocation#34] [Q&A]
I have capacitor Ionic application and I need background geolocation for it. I really appreciate your work guys. But I have some problems with integrating the plugin in my app.

Does it work with capacitor (and 3.0)?
What versions should I use to make it work?

For now I have capcitor 3.0 and:
```
targetSdkVersion 28
compileSdkVersion 28

```
And I get the following error during build: 

`C:\freemind\mobile-app\mobile-app\android\capacitor-cordova-android-plugins\build\intermediates\library_manifest\debug\AndroidManifest.xml:53:9-57:56: AAPT: error: attribute android:foregroundServiceType not found.
`

**Comment by HarelM** [comment]
You can try and use the original plugin (mauron85's), but this plugin should meet the current google requirements which are targetSdk 29.
See here:
https://developer.android.com/distribute/best-practices/develop/target-sdk
Or you can simply fork this one and remove the relevant line from the manifest.
I suggest to progress the targetSdkVersion to 29.
If this is not possible I can help addressing the problem you are facing.

**Reply by MikhailFokanov** [reply]
If I change the version to 29, I get a lot of (48) compile errors. It seems that in v. 29 the package "v4" was removed.
The errors are like:
LocationServiceImpl.java:33: error: package android.support.v4.content does not exist
import android.support.v4.content.LocalBroadcastManager;

Did you have a chance to run it as a compositor plugin for some version of android?

**Reply by HarelM** [reply]
I'm not using Capacitor unfortunately. What's your minSdkVersion? compileSdkVersion? targetSdkVersion?
Can you play with those to make the compilation pass?
You can also probably open android studio and try playing with these configurations there.
I can assure you I have a version that compiles (Cordova not Capacitor), not sure about compile version but I'm using minSdkVersion=22, targetSdkVersion=29

**Reply by MikhailFokanov** [reply]
Thanks for reply. I was playing with version numbers but nothing helped. Currently my configuration is:

    compileSdkVersion 29
    defaultConfig {
        applicationId "io.ionic.starter"
        minSdkVersion 22
        targetSdkVersion 29
        versionCode 1
        versionName "1.0"
        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
    }

Should I set 
```
android.useAndroidX=true
android.enableJetifier=true
```
or to false?

**Reply by MikhailFokanov** [reply]
I think Ionic most often is used with Capacitor. You could try to add the plugin to https://github.com/ionic-team/ionic-react-conference-app and see how it goes for you. 
To add the plugin just do:

```
ionic cap add android (if it wasn't added before)
npm install cordova-background-geolocation-plugin
ionic cap sync
ionic cap run android
```

**Reply by HarelM** [reply]
I'm not sure, but I think capacitor is using androidX libraries, so it's not possible to not use androidX and this library still doesn't support androidX (see #15) yet.
I'm also not an expert with android development so I might be very wrong here.
You can try and create a pull request with all changes required to pass build with androidX I guess...

**Reply by RaddishIoW** [reply]
I use a plugin to enable AndroidX on my cordova project: https://github.com/dpa99c/cordova-plugin-androidx-adapter

**Reply by HarelM** [reply]
Interesting... I'll take a look at it as there are unmaintained plugins which I wasn't very keen to try and upgrade to androidX and this could save me some time and frustration...

**Comment by HarelM** [comment]
I've just released version 2.0 which supports AndroidX. I hope this will solve this issue...

**Comment by Ste-denko** [comment]
Hi All,

@MikhailFokanov did you get the plugin running?
Is it working for you with capacitor?
I also tried to get it working with capacitor but had no luck.
Also opened a [discussion](https://github.com/HaylLtd/cordova-background-geolocation-plugin/discussions/54) for that.

Thanks and BR
